File: js/LINKEDList.ts
class NODE {
    data: number;
    next: NODE | null;

    constructor(value: number) {
        this.data = value;
        this.next = null;
    }
}

class LinkedList {
    head: NODE | null;

    constructor() {
        this.head = null;
    }

    insertEnd(value: number): void {
        const newNode = new NODE(value);

        if (this.head === null) {
            this.head = newNode;
            return;
        }

        let temp = this.head;

        while (temp.next !== null) {
            temp = temp.next;
        }

        temp.next = newNode;
    }

    printList(): void {
        let temp = this.head;

        while (temp !== null) {
            console.log("-->", temp.data);
            temp = temp.next;
        }

        console.log("NULL");
    }
}

const list = new LinkedList();
list.insertEnd(10);
list.insertEnd(20);
list.insertEnd(30);
list.insertEnd(40);
list.printList();

File: c/LinkedList.c
#include <stdio.h>
#include <stdlib.h>

//node structure
struct Node {
    int data;
    struct Node* next;
};

// create  a new node 
struct Node* createNode(int value) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));

    if (newNode == NULL) {
        printf("Memory allocator falied\n");
        exit(1);
    }

    newNode->data = value;
    newNode->next = NULL;
    return newNode;
};

struct Node* insertEnd(struct Node*  head, int value){
    struct Node* newNode = createNode(value);
    // if ther eis not element in node
    if (head == NULL) {
        return newNode;
    }
    struct Node* temp = head;
    while (temp->next != NULL) {
        temp = temp->next;
    }
    temp->next = newNode;
    return head;
};

void printList( struct Node* head) {
    struct Node* temp= head;

    while(temp != NULL) {
        printf("%d ->", temp->data);
        temp = temp->next;
    }
    printf("NULL\n");
}


int main()
{

    printf("**LINKED LIST**\n");

    struct Node* head = NULL;

    head = insertEnd(head,10);
    head = insertEnd(head,20);
    head = insertEnd(head,30);
    printList(head);

    return 0;
}
